### DISCUSSION TITLE
Flux Kontext Validation

### DISCUSSION THREAD

**Discussion by YoadTew** [bghira/SimpleTuner#1630] [Q&A]
Hey,

I'm trying to use the repo to train a FLUX Kontext, and I want to also display validation output every 250 steps.
Inside my `multidatabackend.json` have my training set which is:
```json
{
      "id": "my-edited-images",
      "type": "local",
      "cache_dir_vae": "cache/vae/flux/kontext/edited-images",
      "instance_data_dir": "/datasets/edited-images",
      "conditioning_data": [
        "my-reference-images" 
      ],
      "resolution": 1024,
      "caption_strategy": "textfile"
    },
    {
      "id": "my-reference-images",
      "type": "local",
      "cache_dir_vae": "cache/vae/flux/kontext/ref-images",
      "instance_data_dir": "/datasets/reference-images",
      "conditioning_type": "reference_strict",
      "resolution": 1024,
      "caption_strategy": null
},
``` 

But I also want to set a **separate** validation set, which the model does not train on and contains images and instructions that I want the model to edit during validation, and save to validation_images/print to wandb:

```json
    {
      "id": "my-val-images",
      "type": "local",
      "cache_dir_vae": "cache/vae/flux/kontext/val-images",
      "instance_data_dir": "/datasets/val-images",
      "conditioning_data": [
        "my-val-images" 
      ],
      "resolution": 1024,
      "caption_strategy": "textfile"
    },
```

I've tried setting it up by setting in `config.json`: `"eval_dataset_id": "my-val-images",` but it doesnt seems to work. 
It seems the validation code ignores `validation_sample_images` that was set at the end of `prepare_validation_prompt_list`.

Here is my config.json for reference:
```json
{
    "resume_from_checkpoint": "latest",
    "data_backend_config": "config/multidatabackend.json",
    "aspect_bucket_rounding": 2,
    "seed": 42,
    "minimum_image_size": 0,
    "disable_benchmark": false,
    "output_dir": "output",
    "lora_type": "standard",
    "lora_rank": 64,
    "max_train_steps": 3000,
    "num_train_epochs": 0,
    "checkpointing_steps": 500,
    "checkpoints_total_limit": 15,
    "attention_mechanism": "diffusers",
    "tracker_project_name": "simpletuner-lora-training",
    "tracker_run_name": "flux-kontext-lora",
    "report_to": "wandb",
    "push_to_hub": false,
    "model_type": "lora",
    "model_family": "flux",
    "model_flavour": "kontext",
    "fuse_qkv_projections": false,
    "train_batch_size": 1,
    "gradient_checkpointing": "true",
    "gradient_checkpointing_interval": 1,
    "caption_dropout_probability": 0.05,
    "resolution_type": "pixel_area",
    "resolution": 1024,
    "validation_seed": 42,
    "validation_steps": 250,
    "validation_resolution": "1024x1024",
    "validation_guidance": 2.5,
    "validation_guidance_rescale": 0.0,
    "validation_num_inference_steps": 28,
    "validation_disable_unconditional": true,
    "eval_dataset_id": "my-val-images",
    "mixed_precision": "bf16",
    "optimizer": "adamw_bf16",
    "learning_rate": "1e-4",
    "lr_scheduler": "constant",
    "lr_warmup_steps": 100,
    "conditioning_multidataset_sampling": "random",
    "validation_torch_compile": "false"
}
```

I guess I'm doing something wrong here, anyone had experience with validation for Kontetx?
